pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -815,14 +812,20 @@ func Convert_v1_ContainerDiskSource_To_api_Disk(volumeName string, _ *v1.Contain
 		Source: &api.DiskSource{},
 	}
 
-	source := containerdisk.GetDiskTargetPathFromLauncherView(diskIndex)
-	if info := c.DisksInfo[volumeName]; info != nil {
-		disk.BackingStore.Format.Type = info.Format
-	} else {
-		return fmt.Errorf("no disk info provided for volume %s", volumeName)
+	backingFile := containerdisk.GetDiskTargetPathFromLauncherView(diskIndex)
+	info, err := containerdisk.GetDiskInfo(backingFile)
```

Comment:
Looking at virt-handler, it seems to me that disksInfo is not populated and passed as an empty map when `SyncVirtualMachine` is called for a Running VM.
The data is c.DisksInfo gets there from `c.DisksInfo = l.disksInfo` in the `generateContext` - but l.disksInfo was populated when the VMI was created...

I know that this code is convoluted. We must rework this. 
But right now, I think calling GetDiskInfo here is not needed. We need to use the cached value of GetDisksInfo here

Reply:
Yes, you're right. It's better to use the cache.

Reply:
Done

Reply:
if virt-launcher only uses the disk information at startup, do you think we can remove the code that sends the disk info from the handler, since older VMs won't rely on it from the gRPC call, if I understand correctly?